### DISCUSSION TITLE
Feature Request: named query parameters

### DISCUSSION THREAD

**Discussion by rgeronimi** [duckdb/duckdb#6335] [Ideas]
In prepared statements, DuckDB seems to support only:

- Sequential query parameters, e.g. in Python:
```
connection.execute(
    "SELECT * FROM my_table WHERE name = ? AND country = ?",
    parameters=(name, country)
)
```

- Numbered query parameters, e.g. in Python
```
connection.execute(
    "SELECT * FROM my_table WHERE name = $1 AND (country = $2 OR alt_country = $2)",
    parameters=(name, country)
)
```

- But not named parameters, e.g. in Python
```
connection.execute(
    "SELECT * FROM my_table WHERE name = :name AND (country = :country OR alt_country = :country)",
    parameters={"name": name, "country": country}
)
```
or any alternative syntax 

This feature request is about adding this missing item above.

Why is it important? Because, in complex SQL queries with lots of SQL code and lots of parameters, it makes the code much more readable and easier to understand and maintain by a data analyst.

**Comment by Mytherin** [comment]
Named parameters are supported:

```py
import duckdb
con = duckdb.connect()
con.execute('CREATE TABLE my_table(n VARCHAR, country VARCHAR, alt_country VARCHAR)')
con.execute("INSERT INTO my_table VALUES ('hello', 'w', 'world')")
con.execute(
    "SELECT * FROM my_table WHERE n = $n AND (country = $country OR alt_country = $country)",
    parameters={"n": 'hello', "country": 'world'}
).df()
```

See  #5611

**Comment by rgeronimi** [comment]
Sorry I did check the previous cases and did not find this one - I must have used the wrong search query. Once upgrading python duckdb to v0.7 the issue disappears indeed. Thanks!
